File: JavaScript/01.JavaScript-Basic-February-2021/11.nestedLoops-Lab/04.sumOfTwoNumbers.js
function sum(input) {
    let a = Number(input[0]);
    let b = Number(input[1]);
    let c = Number(input[2]);
    let combinationCount = 0;

    let flag = false;
    for (let i = a; i <= b; i++) {
        for (let j = a; j <= b; j++) {
            combinationCount++;
            if (i + j === c) {
                console.log(`Combination N:${combinationCount} (${i} + ${j} = ${i + j})`);
                flag = true;
                break;
            }
        }
        if (flag) {
            break;
        }
    }
    if (!flag) {
        console.log(`${combinationCount} combinations - neither equals ${c}`);
    }
}

sum(["1", "10", "5"]);
// sum(["88", "888", "1000"]);
// sum(["88", "888", "1000"]);
// sum(["88", "888", "2000"]);

File: Python/01.Python-Basic-April-2019/12.nested_loops_lab/04.sum_of_two_numbers.py
start_interval = int(input())
end_interval = int(input())
magic_number = int(input())

combinations = 0
has_magic_number = False

for x in range(start_interval, end_interval + 1):
    for y in range(start_interval, end_interval + 1):
        combinations += 1

        if (x + y) == magic_number:
            has_magic_number = True
            break

    if has_magic_number:
        break

if has_magic_number:
    print(f'Combination N:{combinations} ({x} + {y} = {magic_number})')
else:
    print(f'{combinations} combinations - neither equals {magic_number}')
